### PR TITLE
telemetry adjustments

### PR DESCRIPTION
**Pull request by johndietz**

### CODE REVIEW

File: charts/kubefirst-api/templates/deployment.yaml [hunk 1]
```diff
@@ -164,17 +164,25 @@ spec:
...
-            "--install-method",
-            {{ .Values.installMethod | default "helm" | quote }},
-            "--kubefirst-team",
-            {{ .Values.kubefirstTeam | default "undefined" | quote }},
-            "--kubefirst-team-info",
-            {{ .Values.kubefirstTeamInfo | default "undefined" }}
           ]
           env:
-            - name: KUBEFIRST_VERSION
+            - name: CLI_VERSION
```
Comment (johndietz) [inline charts/kubefirst-api/templates/deployment.yaml]:
```suggestion
            - name: KUBEFIRST_VERSION
```

File: internal/controller/argocd.go [hunk 2]
```diff
@@ -66,13 +59,13 @@ func (clctrl *ClusterController) InstallArgoCD() error {
...
+
+			telemetry.SendCountMetric(clctrl.Telemetry, metrics.ArgoCDInstallFailed, err.Error())
+
 			return err
 		}
-		telemetryShim.Transmit(segmentClient, segment.MetricArgoCDInstallCompleted, "")
 
-		telemetryShim.Transmit(segmentClient, segment.MetricArgoCDInstallStarted, "")
-		telemetryShim.Transmit(segmentClient, segment.MetricArgoCDInstallCompleted, "")
+		telemetry.SendCountMetric(clctrl.Telemetry, metrics.ArgoCDInstallCompleted, err.Error())
```
Comment (johndietz) [inline internal/controller/argocd.go]: some data has been lost here - let's talk

File: charts/kubefirst-api/templates/deployment.yaml (hunk 1 above)
Comment (jarededwards) [inline charts/kubefirst-api/templates/deployment.yaml]:
```suggestion
            - name: KUBEFIRST_VERSION
```

File: internal/controller/argocd.go [hunk 3]
```diff
@@ -185,12 +178,7 @@ func (clctrl *ClusterController) DeployRegistryApplication() error {
 		return err
 	}
 
-	// Telemetry handler
-	segmentClient, err := telemetryShim.SetupTelemetry(cl)
-	if err != nil {
-		return err
-	}
-	defer segmentClient.Client.Close()
+	// telemetry todo
```
Comment (jarededwards) [inline internal/controller/argocd.go]:
```suggestion

```

File: providers/google/delete.go [hunk 4]
```diff
@@ -259,7 +252,7 @@ func DeleteGoogleCluster(cl *pkgtypes.Cluster) error {
 		}
 	}
 
-	telemetryShim.Transmit(segmentClient, segment.MetricClusterDeleteCompleted, "")
+	//telemetry.Transmit(segmentClient, telemetry.MetricClusterDeleteCompleted, "")
```
Comment (johndietz) [inline providers/google/delete.go]: needs deleted?

File: providers/vultr/delete.go [hunk 5]
```diff
@@ -43,19 +43,12 @@ func DeleteVultrCluster(cl *pkgtypes.Cluster) error {
...
 
-	// Telemetry handler
-	segmentClient, err := telemetryShim.SetupTelemetry(*cl)
-	if err != nil {
-		return err
-	}
-	defer segmentClient.Client.Close()
-
-	telemetryShim.Transmit(segmentClient, segment.MetricClusterDeleteStarted, "")
+	telemetry.SendCountMetric(segmentClient, metrics.ClusterInstallStarted, "")
```
Comment (johndietz) [inline providers/vultr/delete.go]: cluster delete started

File: providers/vultr/delete.go [hunk 6]
```diff
@@ -287,7 +280,7 @@ func DeleteVultrCluster(cl *pkgtypes.Cluster) error {
 		}
 	}
 
-	telemetryShim.Transmit(segmentClient, segment.MetricClusterDeleteCompleted, "")
+	//telemetry.Transmit(segmentClient, telemetry.MetricClusterDeleteCompleted, "")
```
Comment (johndietz) [inline providers/vultr/delete.go]: needs delete?

File: internal/controller/state.go [hunk 7]
```diff
@@ -132,8 +126,8 @@ func (clctrl *ClusterController) StateStoreCredentials() error {
 			_, err := clctrl.GoogleClient.CreateBucket(clctrl.KubefirstStateStoreBucketName, []byte(clctrl.GoogleAuth.KeyFile))
 			if err != nil {
 				msg := fmt.Sprintf("error creating google bucket %s: %s", clctrl.KubefirstStateStoreBucketName, err)
-				log.Error(msg)
-				telemetryShim.Transmit(segmentClient, segment.MetricStateStoreCreateFailed, msg)
+				log.Error()
+				telemetry.SendCountMetric(clctrl.Telemetry, metrics.StateStoreCreateFailed, msg)
```
Comment (jarededwards) [inline internal/controller/state.go]:
```suggestion
				telemetry.SendCountMetric(clctrl.Telemetry, metrics.StateStoreCredentialsCreateFailed, msg)
```

File: internal/controller/state.go [hunk 8]
```diff
@@ -148,7 +142,7 @@ func (clctrl *ClusterController) StateStoreCredentials() error {
 
 			objst, err := vultrConf.CreateObjectStorage(clctrl.KubefirstStateStoreBucketName)
 			if err != nil {
-				telemetryShim.Transmit(segmentClient, segment.MetricStateStoreCreateFailed, err.Error())
+				telemetry.SendCountMetric(clctrl.Telemetry, metrics.StateStoreCreateFailed, err.Error())
```
Comment (jarededwards) [inline internal/controller/state.go]:
```suggestion
				telemetry.SendCountMetric(clctrl.Telemetry, metrics.StateStoreCredentialsCreateFailed, err.Error())
```

File: providers/google/delete.go (hunk 4 above)
Comment (johndietz) [inline providers/google/delete.go]:
```suggestion
	telemetry.Transmit(segmentClient, telemetry.MetricClusterDeleteCompleted, "")
```

File: providers/vultr/delete.go (hunk 5 above)
Comment (johndietz) [inline providers/vultr/delete.go]:
```suggestion
	telemetry.SendCountMetric(segmentClient, metrics.ClusterDeleteStarted, "")
```

**Review by jarededwards — APPROVED** [review]

**Pull request merged**